pints/_mcmc/__init__.py

```diff
@@ -543,39 +546,70 @@ def run(self):
 
             # Get points
             if self._single_chain:
-                xs = [sampler.ask() for sampler in self._samplers]
+                # First step: create array for initial samples
+                if evaluations == 0:
+                    xs = [sampler.ask() for sampler in self._samplers]
+                # Subsequent steps: ask for log_pdf of points only in active
+                # samplers
+                else:
+                    for i, sampler in enumerate(self._samplers):
+                        if sampler._active:
+                            xs[i] = sampler.ask()
             else:
                 xs = self._samplers[0].ask()
 
             # Calculate logpdfs
             fxs = evaluator.evaluate(xs)
```

Comment:
You're re-evaluating the same `x` for inactive samplers here. That's potentially very costly so better to avoid that!

I would suggest doing this a different way. For example, you could keep an array of indices of active samplers, then
```
for i, sampler in enumerate(self._sampler):
    if sampler._active:
        ...
```
just becomes
```
for i in i_active_samplers:
    ...
```
and the number of active samplers is `len(i_active_samplers)`